src/main/scala/com/codacy/client/stash/service/RepositoryServices.scala

```diff
@@ -74,6 +74,22 @@ class RepositoryServices(client: StashClient) {
       )
   }
 
+  def getRepositoryGroups(
+      projectKey: String,
+      repositorySlug: String,
+      pageRequest: Option[PageRequest]
+  ): RequestResponse[Seq[Group]] = pageRequest match {
+    case Some(pageRequest) =>
+      client.executePaginated(
+        Request(s"$BASE/$projectKey/repos/$repositorySlug/permissions/groups", classOf[Seq[Group]]),
+        pageRequest
+      )
+    case None =>
+      client.executePaginated(
+        Request(s"$BASE/$projectKey/repos/$repositorySlug/permissions/groups", classOf[Seq[Group]])
```

Comment:
It seems a bit of a shame that we repeat the "Request string" every timen it seems that the executePaginated method should receive an option

Reply:
Its overloaded as to not break all the other implementations throughout the code. At this point in time I'm alright with it being like this.